articles/how-to-fix-and-prevent-sql-injection-in-wordpress/index.md

```diff
@@ -0,0 +1,194 @@
+~~~~ sql
+“and 1=1”
+~~~~
+~~~~ sql
+“and 1=2”
+~~~~
+
+If the application works as expected on the first clause but works anomaly with the second clause, it is vulnerable to an SQLi.
+
+#### Time-based SQLi
+
+This attack is used to determine if a vulnerability exists in an SQL database. A pre-defined time function is put in place by an attacker targeting the database management system. For instance, the function sleep() in MySQL instructs a database to wait for a number of seconds. See below;
+~~~~sql
+select * from comments WHERE post_id=1-SLEEP(15);
+~~~~ 
+
+if the query results in a delay, then the database is vulnerable to an attack.
+
+### In-Band SQLi
+
+This is by far the simplest form of SQLi. An attacker uses the same channel to input malicious SQL codes as well gather results on the same. Let's look into twotypes of In-Band SQLi namely: Error-based SQLi and Union-based SQLi.
+
+#### Error- Based Injection
+
+This type of an attack is used in the earlier phases of SQLi. The attacker uses the error-based injection to learn more about the database structure and table names as used in the web application. For example, once an attack has been initiated the corresponding error messages may include table names and column table names. The information can be useful to further the attack.
+
+#### Union Based Injection
+
+This type of injection uses the [SQL UNION command](https://www.w3schools.com/sql/sql_union.asp#:~:text=The%20SQL%20UNION%20Operator,also%20have%20similar%20data%20types) to append or combine requests which are then displayed as a partial HTTP response. This type of SQLi attack targets a band where the UNION operator is able to simultaneously extract data from several table with the database. Besides, for the attack to be successful the table must have the same type of data and same number of columns. Hackers manipulate the SELECT statement to collect the information they are looking out for.
+
+#### WordPress Plugins Vulnerability
+
+WordPress is the most popular open-source Content Management System (CMS) running millions of websites. It's approximated the [CMS enjoys a 61% market share](https://kinsta.com/blog/wordpress-statistics/). Your WordPress website is secure from SQLi only if the WordPress core files are up-to-date.
+
+##### What are WordPress Core files?
+
+This are the files that make up the open-source CMS project. Below is an image of what [WordPress core files](https://wpsupergeek.com/what-are-the-wordpress-core-files/#:~:text=When%20you%20install%20WordPress%2C%20you,admin%20interface%20of%20your%20website.&amp;text=Core%20files%20make%20up%20the,themes%2C%20plugins%2C%20and%20images.) look-like.
```

Comment:
These not This

look like are two seperate words.

Reply:
This still needs to be corrected, @jkw19.

Reply:
@louisefindlay23 Corrected.